low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -216,10 +229,11 @@ def properties_changed_handler(self, interface, changed_properties,
 
         if self.state != self.previous_state:
             if self.state == "active":
-                if self._service_state is FailedState:
-                    Service.alerts.put(
-                        ServiceMonitor.get_alert(self, ResolvedAlert))
-                self.new_service_state(ActiveState)
+                if self._service_state is FailedState and self.name not in Service.active_services:
+                    self.new_service_state(ActiveState)
+                    Service.active_services.add(self.name)
+                if self._service_state is InactiveState:
+                    self.new_service_state(ActiveState)
```

Comment:
> Service.active_services.add(self.name)

Adding to active service needed here also i believe @line 232
>  if self._service_state in (FailedState, InactiveState) and self.name not in Service.active_services:

In which case, can easily club the condition i believe

Reply:
No, Here don't need to add service in active_service list, because service will be in InactiveState only till threshold_time after that service_state change to FailedState, then we raise fault alert.

But if service state change from inactive to active before threshold time, then we will not raise any fault alert, and hence fault_resolved alert also not needed, thats why service not added in active_service list.

Reply:
oh right, agree. Guess we are good.